### PR TITLE
remove dependence on and references to db docker definition in `delphi/operations` repo

### PR DESCRIPTION
**Pull request by melange396**
- removes building of `delphi_database` from `ci.yaml` and dev `Makefile`
- updates related documentation
- also annotates `dev/docker/python/Dockerfile` with a reference to its dependency on `delphi/operations` repository 

we havent needed `dev/docker/database/Dockerfile` from `delphi/operations` (or the `delphi_database` image it builds) since the v4 merge, which moved to building on a Percona container.

### CODE REVIEW

File: dev/local/Makefile
Comment (dshemetov) [inline dev/local/Makefile:81]: til: lifehack! productivity +100% 👍

**Review by dshemetov — DISMISSED** [review]
lgtm, happy to see our dependencies and Dockerfiles slowly get unified, instead of being split across multiple repos

**Comment by dshemetov** [comment]
tag #965

**Comment by melange396** [comment]
nice, didnt realize you had that issue open regarding the `operations` repo [in actuality, it was probably less "didnt realize" and more "forgot"]

**Comment by melange396** [comment]
ugh, i made a mess of the commit history, but this should now be against the `dev` branch instead of `main`. :(

**Review by dshemetov — APPROVED** [review]
diffs look the same as before, approving again

**Comment by dshemetov** [comment]
you can rebase, clean, and force push, if you want. i didn't check this PR locally

**Review by krivard — APPROVED** [review]
👍 j'approve

**Comment by melange396** [comment]
whoops, i let this languish for a month and a half, gonna merge now

**Comment by melange396** [comment]
...or rather, i will merge when i get another approval

**Comment by melange396** [comment]
this also furthers the goals of #965

**Review by dshemetov — APPROVED** [review]

**Comment by melange396** [comment]
fwiw, when i fixed the merge conflicts on this, i went with the `dev` branch for all of them

**Pull request merged**
